lib/Sema/CSApply.cpp

```diff
@@ -7601,67 +7601,147 @@ Expr *ExprRewriter::finishApply(ApplyExpr *apply, Type openedType,
   }
 
   // We have a type constructor.
-  auto metaTy = cs.getType(fn)->castTo<AnyMetatypeType>();
-  auto ty = metaTy->getInstanceType();
-
-  if (!cs.isTypeReference(fn)) {
-    bool isExistentialType = false;
-    // If this is an attempt to initialize existential type.
-    if (auto metaType = cs.getType(fn)->getAs<MetatypeType>()) {
-      auto instanceType = metaType->getInstanceType();
-      isExistentialType = instanceType->isExistentialType();
-    }
+  if (auto metaTy = cs.getType(fn)->getAs<AnyMetatypeType>()) {
```

Comment:
Note: the diff at this part of the file isn't actually that big.
I simply moved the metatype-handling code (lines 7604-7655) into an if-block.

Reply:
This method is already quite long. Can you split off this part into a separate method?